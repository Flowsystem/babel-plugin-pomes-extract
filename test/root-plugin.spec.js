const fs = require('fs');
const path = require('path');
const babel = require('@babel/core');

const plugin = (filename) => babel.transformFileSync(path.join(__dirname, 'fixtures', filename), {
  presets: [
    '@babel/preset-react',
  ],
  plugins: [
    [
      path.join(__dirname, '../index.js'),
      {
        outputDir: path.join(__dirname, 'gettext/resources'),
        headers: {
          'po-revision-date': '2019-08-13T07:45:01.119Z',
        },
      },
      '@oneflowab/babel-plugin-pomes-extract',
    ],
  ],
});

const readPotFile = (filename) => fs.readFileSync(path.join(__dirname, 'gettext/resources/test/fixtures', filename), { encoding: 'utf8' });

describe('Pomes Extract Plugin', () => {
  describe('#message', () => {
    it('extract a message call', () => {
      plugin('message.js');

      const potFileContent = readPotFile('message.js.pot');

      expect(potFileContent.toString()).toMatchSnapshot();
    });

    it('extract a message call with plural form', () => {
      plugin('message-with-plural.js');

      const potFileContent = readPotFile('message-with-plural.js.pot');

      expect(potFileContent.toString()).toMatchSnapshot();
    });

    it('merge a message with singular and plural forms', () => {
      plugin('message-with-singular-and-plural.js');

      const potFileContent = readPotFile('message-with-singular-and-plural.js.pot');

      expect(potFileContent.toString()).toMatchSnapshot();
    });

    xit('raise warning for message without comment', () => {
      const oldLog = console.log;
      console.log = jest.fn();
      plugin('message-without-comment.js');

      const potFileContent = readPotFile('message-without-comment.js.pot');

      expect(potFileContent.toString()).toMatchSnapshot();

      expect(console.log.mock.calls).toMatchSnapshot();
      console.log = oldLog;
    });
  });

  describe('<Message />', () => {
    it('extract a Message component', () => {
      plugin('message-component.js');

      const potFileContent = readPotFile('message-component.js.pot');

      expect(potFileContent.toString()).toMatchSnapshot();
    });

    it('extract a message call with plural form', () => {
      plugin('message-component-with-plural.js');

      const potFileContent = readPotFile('message-component-with-plural.js.pot');

      expect(potFileContent.toString()).toMatchSnapshot();
    });

    it('merge a message with singular and plural forms', () => {
      plugin('message-component-with-singular-and-plural.js');

      const potFileContent = readPotFile('message-component-with-singular-and-plural.js.pot');

      expect(potFileContent.toString()).toMatchSnapshot();
    });
  });
});
