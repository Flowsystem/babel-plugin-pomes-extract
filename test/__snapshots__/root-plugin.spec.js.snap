// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pomes Extract Plugin #message extract a message call 1`] = `
"msgid \\"\\"
msgstr \\"\\"
\\"Content-Type: text/plain; charset=utf-8\\\\n\\"
\\"Plural-Forms: nplurals=2; plural=(n!=1);\\\\n\\"
\\"Language: en_US\\\\n\\"
\\"PO-Revision-Date: 2019-08-13T07:45:01.119Z\\\\n\\"

#: test/fixtures/message.js:3
#. bar
msgid \\"foo\\"
msgstr \\"\\""
`;

exports[`Pomes Extract Plugin #message extract a message call with plural form 1`] = `
"msgid \\"\\"
msgstr \\"\\"
\\"Content-Type: text/plain; charset=utf-8\\\\n\\"
\\"Plural-Forms: nplurals=2; plural=(n!=1);\\\\n\\"
\\"Language: en_US\\\\n\\"
\\"PO-Revision-Date: 2019-08-13T07:45:01.119Z\\\\n\\"

#: test/fixtures/message-with-plural.js:4
#. bar comment
msgid \\"foo message\\"
msgid_plural \\"foo message plural\\"
msgstr[0] \\"\\"
msgstr[1] \\"\\""
`;

exports[`Pomes Extract Plugin #message merge a message with singular and plural forms 1`] = `
"msgid \\"\\"
msgstr \\"\\"
\\"Content-Type: text/plain; charset=utf-8\\\\n\\"
\\"Plural-Forms: nplurals=2; plural=(n!=1);\\\\n\\"
\\"Language: en_US\\\\n\\"
\\"PO-Revision-Date: 2019-08-13T07:45:01.119Z\\\\n\\"

#: test/fixtures/message-with-singular-and-plural.js:4
#: test/fixtures/message-with-singular-and-plural.js:10
#. bar comment (singular)
#. bar comment (plural)
msgid \\"foo message\\"
msgid_plural \\"foo message plural\\"
msgstr[0] \\"\\"
msgstr[1] \\"\\""
`;

exports[`Pomes Extract Plugin #message raise warning for message without comment 1`] = `
"msgid \\"\\"
msgstr \\"\\"
\\"Content-Type: text/plain; charset=utf-8\\\\n\\"
\\"Plural-Forms: nplurals=2; plural=(n!=1);\\\\n\\"
\\"Language: en_US\\\\n\\"
\\"PO-Revision-Date: 2019-08-13T07:45:01.119Z\\\\n\\"

#: test/fixtures/message-without-comment.js:3
msgid \\"foo\\"
msgstr \\"\\""
`;

exports[`Pomes Extract Plugin #message split messages with context 1`] = `
"msgid \\"\\"
msgstr \\"\\"
\\"Content-Type: text/plain; charset=utf-8\\\\n\\"
\\"Plural-Forms: nplurals=2; plural=(n!=1);\\\\n\\"
\\"Language: en_US\\\\n\\"
\\"PO-Revision-Date: 2019-08-13T07:45:01.119Z\\\\n\\"

#: test/fixtures/message-with-singular-plural-and-context.js:4
#. bar comment (singular)
msgid \\"foo message\\"
msgstr \\"\\"

#: test/fixtures/message-with-singular-plural-and-context.js:9
#. bar comment (singular)
msgctxt \\"singular context\\"
msgid \\"foo message\\"
msgstr \\"\\"

#: test/fixtures/message-with-singular-plural-and-context.js:15
#. bar comment (plural)
msgctxt \\"plural context\\"
msgid \\"foo message\\"
msgid_plural \\"foo message plural\\"
msgstr[0] \\"\\"
msgstr[1] \\"\\""
`;

exports[`Pomes Extract Plugin <Message /> extract a Message component 1`] = `
"msgid \\"\\"
msgstr \\"\\"
\\"Content-Type: text/plain; charset=utf-8\\\\n\\"
\\"Plural-Forms: nplurals=2; plural=(n!=1);\\\\n\\"
\\"Language: en_US\\\\n\\"
\\"PO-Revision-Date: 2019-08-13T07:45:01.119Z\\\\n\\"

#: test/fixtures/message-component.js:7
#. bar
msgid \\"foo\\"
msgstr \\"\\""
`;

exports[`Pomes Extract Plugin <Message /> extract a Message component with plural form 1`] = `
"msgid \\"\\"
msgstr \\"\\"
\\"Content-Type: text/plain; charset=utf-8\\\\n\\"
\\"Plural-Forms: nplurals=2; plural=(n!=1);\\\\n\\"
\\"Language: en_US\\\\n\\"
\\"PO-Revision-Date: 2019-08-13T07:45:01.119Z\\\\n\\"

#: test/fixtures/message-component-with-plural.js:1
#. bar
msgid \\"foo\\"
msgid_plural \\"foo {n}\\"
msgstr[0] \\"\\"
msgstr[1] \\"\\""
`;

exports[`Pomes Extract Plugin <Message /> merge a Message component with singular and plural forms 1`] = `
"msgid \\"\\"
msgstr \\"\\"
\\"Content-Type: text/plain; charset=utf-8\\\\n\\"
\\"Plural-Forms: nplurals=2; plural=(n!=1);\\\\n\\"
\\"Language: en_US\\\\n\\"
\\"PO-Revision-Date: 2019-08-13T07:45:01.119Z\\\\n\\"

#: test/fixtures/message-component-with-singular-and-plural.js:2
#: test/fixtures/message-component-with-singular-and-plural.js:7
#. bar comment (singular)
#. bar comment (plural)
msgid \\"foo message\\"
msgid_plural \\"foo message plural\\"
msgstr[0] \\"\\"
msgstr[1] \\"\\""
`;

exports[`Pomes Extract Plugin <Message /> raise warning for Message component without comment 1`] = `
"msgid \\"\\"
msgstr \\"\\"
\\"Content-Type: text/plain; charset=utf-8\\\\n\\"
\\"Plural-Forms: nplurals=2; plural=(n!=1);\\\\n\\"
\\"Language: en_US\\\\n\\"
\\"PO-Revision-Date: 2019-08-13T07:45:01.119Z\\\\n\\"

#: test/fixtures/message-component-without-comment.js:1
msgid \\"foo\\"
msgstr \\"\\""
`;

exports[`Pomes Extract Plugin <Message /> split Messages with context 1`] = `
"msgid \\"\\"
msgstr \\"\\"
\\"Content-Type: text/plain; charset=utf-8\\\\n\\"
\\"Plural-Forms: nplurals=2; plural=(n!=1);\\\\n\\"
\\"Language: en_US\\\\n\\"
\\"PO-Revision-Date: 2019-08-13T07:45:01.119Z\\\\n\\"

#: test/fixtures/message-component-with-singular-plural-and-context.js:2
#. bar comment (singular)
msgid \\"foo message\\"
msgstr \\"\\"

#: test/fixtures/message-component-with-singular-plural-and-context.js:6
#. bar comment (singular)
msgctxt \\"singular context\\"
msgid \\"foo message\\"
msgstr \\"\\"

#: test/fixtures/message-component-with-singular-plural-and-context.js:11
#. bar comment (plural)
msgctxt \\"plural context\\"
msgid \\"foo message\\"
msgid_plural \\"foo message plural\\"
msgstr[0] \\"\\"
msgstr[1] \\"\\""
`;
